### PR TITLE
use new kots linter

### PR DESCRIPTION
**Pull request by salahalsaleh**

### CODE REVIEW

File: cli/cmd/release_lint.go [hunk 1]
```diff
@@ -89,9 +49,19 @@ func (r *runners) releaseLintStable(cmd *cobra.Command, args []string) error {
...
-	if err != nil {
-		return errors.Wrap(err, "read yaml dir")
+	var lintReleaseYAML []byte
+	if r.appType == "kots" {
+		b, err := tarYAMLDir(r.args.lintReleaseYamlDir)
+		if err != nil {
+			return errors.Wrap(err, "failed to read yaml dir")
+		}
+		lintReleaseYAML = b
+	} else {
```
Comment (dexhorthy) [inline cli/cmd/release_lint.go:59]: I don't really understand this -- for Platform, linting is not supported, and for ship there's no Dir, its just a single file

File: client/release.go [hunk 2]
```diff
@@ -155,26 +155,18 @@ func (c *Client) PromoteRelease(appID string, appType string, sequence int64, la
 	return errors.New("unknown app type")
 }
 
-// yamlOrJSON is either a single ship yaml file, or a serialized JSON object describing a yaml-dir, created by readYAMLDir()
+// data is either a single ship yaml file, or a serialized JSON object describing a yaml-dir, created by readYAMLDir()
```
Comment (dexhorthy) [inline client/release.go:158]: what was the reasoning behind this naming change? I had switched it from `data` because `data` was kind of unclear.

Comment (salahalsaleh) [reply client/release.go:158]: for kots, it is no longer a yaml or a json string, it is the tarred yaml dir containing all of the manifests as a []byte.

File: cli/cmd/release_lint.go (hunk 1 above)
Comment (salahalsaleh) [reply cli/cmd/release_lint.go:59]: This was very confusing for me as well, but i did not change any functionality. I simple eliminated the old route for kots linting.

Comment (salahalsaleh) [reply cli/cmd/release_lint.go:59]: Check out https://github.com/replicatedhq/replicated/pull/132/files#diff-4a4b54d68a6f920d2d4433fa0a656336L92

File: client/release.go (hunk 2 above)
Comment (dexhorthy) [reply client/release.go:158]: okay, wanna update the comment? I'm still trying to understand how this is gonna work for ship

Comment (dexhorthy) [reply client/release.go:158]: @salahalsaleh it appears I removed all support for ship linting in this PR 🤦  https://github.com/replicatedhq/replicated/commit/bb7463d1145be607b4eec4f8f8d4d377398e6999

Comment (salahalsaleh) [reply client/release.go:158]: Aha, so we do not support ship linting anymore? I can clean all of this up????? :)

Comment (dexhorthy) [reply client/release.go:158]: yeah lets

File: client/release.go [hunk 3]
```diff
@@ -155,26 +155,15 @@ func (c *Client) PromoteRelease(appID string, appType string, sequence int64, la
...
+// data is a []byte describing a tarred yaml-dir, created by tarYAMLDir()
 // this Client abstraction continue to spring more leaks :)
-func (c *Client) LintRelease(appID string, appType string, yamlOrJSON string) ([]types.LintMessage, error) {
+func (c *Client) LintRelease(appType string, data []byte) ([]types.LintMessage, error) {
 
 	if appType == "platform" {
 		return nil, errors.New("Linting is not yet supported in this CLI, please install github.com/replicatedhq/replicated-lint to lint this application")
-		// return c.PlatformClient.LintRelease(appID, yamlOrJSON)
-	} else if appType == "ship" {
-		return c.ShipClient.LintRelease(appID, yamlOrJSON)
```
Comment (dexhorthy) [inline client/release.go:166]: Rather than removing this branch, can we return an error about how Ship linting is not supported?

Comment (salahalsaleh) [reply client/release.go:166]: yep, for sure

**Pull request merged**